src/Skoruba.IdentityServer4.STS.Identity/Helpers/IdentityServerBuilderExtensions.cs

```diff
@@ -0,0 +1,125 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Extensions.Configuration;
+using Skoruba.IdentityServer4.STS.Identity.Configuration;
+
+namespace Skoruba.IdentityServer4.STS.Identity.Helpers
+{
+    public static class IdentityServerBuilderExtensions
+    {
+        private const string CertificateNotFound = "Certificate not found";
+        private const string SigningCertificateThumbprintNotFound = "Signing certificate thumbprint not found";
+        private const string SigningCertificatePathIsNotSpecified = "Signing certificate file path is not specified";
+
+        private const string ValidationCertificateThumbprintNotFound = "Validation certificate thumbprint not found";
+        private const string ValidationCertificatePathIsNotSpecified = "Validation certificate file path is not specified";
+
+        /// <summary>
+        /// Add custom signing certificate from certification store according thumbprint or from file
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="configuration"></param>
+        /// <returns></returns>
+        public static IIdentityServerBuilder AddCustomSigningCredential(this IIdentityServerBuilder builder, IConfiguration configuration)
+        {
+            var certificateConfiguration = configuration.GetSection(nameof(CertificateConfiguration)).Get<CertificateConfiguration>();
+
+            if (certificateConfiguration.UseSigningCertificateThumbprint)
+            {
+                if (string.IsNullOrWhiteSpace(certificateConfiguration.SigningCertificateThumbprint))
+                {
+                    throw new Exception(SigningCertificateThumbprintNotFound);
+                }
+
+                var certStore = new X509Store(StoreName.My, StoreLocation.LocalMachine);
+                certStore.Open(OpenFlags.ReadOnly);
+
+                var certCollection = certStore.Certificates.Find(X509FindType.FindByThumbprint, certificateConfiguration.SigningCertificateThumbprint, true);
+                if (certCollection.Count == 0)
+                {
+                    throw new Exception(CertificateNotFound);
+                }
+
+                var certificate = certCollection[0];
+
+                builder.AddSigningCredential(certificate);
+            }
+            else if (certificateConfiguration.UseSigningCertificateFile)
+            {
+                if (string.IsNullOrWhiteSpace(certificateConfiguration.SigningCertificateFilePath))
+                {
+                    throw new Exception(SigningCertificatePathIsNotSpecified);
+                }
+
+                if (File.Exists(certificateConfiguration.SigningCertificateFilePath))
+                {
+                    builder.AddSigningCredential(new X509Certificate2(certificateConfiguration.SigningCertificateFilePath, certificateConfiguration.SigningCertificateFilePassword));
```

Comment:
I think CryptographicException should be catched. It's thrown when the certificate is invalid or the password is incorrect.

Reply:
Great catch, I will fix it.

Reply:
1. Check if certificate file exists - the exception thrown by X509Certificate2 is not intuitive particularly on Linux versions.
2. After creating certificate check IsPrivateKey property - it is required for signing.
3, Catch exception mentioned by @svendu

Reply:
I fixed it - please take a look. Thanks!